src/factory/ChartFactory.js

```diff
@@ -251,7 +247,8 @@ function ChartFactory({
         domainPadding.x[0] +
         domainPadding.x[1];
       const height = visualProps.height || theme.bar.height;
-      const computedWidth = horizontal ? height : computedSize;
+      const computedWidth =
+        horizontal || computedSize < height ? height : computedSize;
```

Comment:
I think:
1. `horizontal` if requested, _or_
2. `horizontal` if `computedSize > width` i.e. it won't fit the `width`.
Screen height is not really limited but width is.

PS: We should do this for `grouped_column` as well (and every other place we're computing the size).

Reply:
@kilemensi  spot on :100: